### PR TITLE
- merge broadcasts if and only if simple item type is broadcast

### PR DESCRIPTION
**Pull request by ljeyaratnam**

### CODE REVIEW

**Comment by tfm** [comment]
There are many whitespace changes here. What happened?

File: src/main/java/org/atlasapi/query/v2/ContentWriteController.java [hunk 1]
```diff
@@ -256,16 +278,36 @@ private Content complexify(Description inputContent) {
...
+                if (Iterables.isEmpty(item.getVersions())) {
+                    item.addVersion(new Version());
+                }
+                Version existingVersion = item.getVersions().iterator().next();
+                Version postedVersion = Iterables.getOnlyElement(update.getVersions());
+                mergeVersions(existingVersion, postedVersion);
+            }
+            return (Content) existing;
+        }
+        throw new IllegalStateException("Entity for " + update.getCanonicalUri() + " not Content");
```
Comment (tfm) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: Turn this into a guard, as it leads to less nesting.

if (!(existing instanceof Item) {
  throw new ...
}

// rest of class

File: src/main/java/org/atlasapi/query/v2/ContentWriteController.java [hunk 2]
```diff
@@ -256,16 +278,36 @@ private Content complexify(Description inputContent) {
...
     }
 
+    private Content mergeBroadcasts(Maybe<Identified> possibleExisting, Content update) {
+        if (possibleExisting.isNothing()) {
+            throw new IllegalStateException("Entity for " + update.getCanonicalUri() + " does not exists");
+        }
+        Identified existing = possibleExisting.requireValue();
+        if (existing instanceof Item) {
+            Item item = (Item) existing;
+            if (!update.getVersions().isEmpty()) {
```
Comment (tfm) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: Why do need this check? Can't we just have a `for` loop over all `Version`s?

File: src/main/java/org/atlasapi/query/v2/ContentWriteController.java [hunk 3]
```diff
@@ -256,16 +278,36 @@ private Content complexify(Description inputContent) {
...
+                if (Iterables.isEmpty(item.getVersions())) {
+                    item.addVersion(new Version());
+                }
+                Version existingVersion = item.getVersions().iterator().next();
+                Version postedVersion = Iterables.getOnlyElement(update.getVersions());
```
Comment (tfm) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: What if there's more than one `Version`

**Comment by ljeyaratnam** [comment]
white space - reformatted the code

**Comment by tfm** [comment]
Always separate reformatting commits from functional changes; otherwise it's difficult to see what's changed. Don't worry about it this time round, but for future reference.

File: src/main/java/org/atlasapi/query/v2/ContentWriteController.java (hunk 2 above)
Comment (ljeyaratnam) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: I followed the existing code pattern to merge just only one versions. Also I am not sure how we can be sure if we were to loop over the update versions and update version and existing version are same

File: src/main/java/org/atlasapi/query/v2/ContentWriteController.java [hunk 4]
```diff
@@ -216,7 +238,7 @@ private Item mergeItems(Item existing, Item update) {
             mergeVersions(existingVersion, postedVersion);
```
Comment (ljeyaratnam) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: As you can see existing code only merges one version

File: src/main/java/org/atlasapi/query/v2/ContentWriteController.java [hunk 5]
```diff
@@ -256,16 +278,41 @@ private Content complexify(Description inputContent) {
...
-    
+
     private Void error(HttpServletResponse response, int code) {
         response.setStatus(code);
         response.setContentLength(0);
         return null;
     }
 
+    private Content mergeBroadcasts(Maybe<Identified> possibleExisting, Content update) {
+        if (possibleExisting.isNothing()) {
```
Comment (tfm) [inline src/main/java/org/atlasapi/query/v2/ContentWriteController.java]: If `possibleExisting` is nothing, we should not throw. We should save a new item. See previous behaviour with the `merge` method.

**Pull request merged**